geopandas/io/tests/test_sql.py

```diff
@@ -585,3 +585,38 @@ def test_row_order(self, engine_postgis, df_nybb):
         sql = "SELECT * FROM {table};".format(table=table)
         df = read_postgis(sql, engine, geom_col="geometry")
         assert df["BoroCode"].tolist() == correct_order
+
+    def test_append_before_table_exists(self, engine_postgis, df_nybb):
+        """
+        Tests that insert works with if_exists='append' when table does not exist yet.
+        """
+        engine = engine_postgis
+
+        table = "nybb"
+        # If table exists, delete it before trying to write with defaults
+        drop_table_if_exists(engine, table)
+
+        write_postgis(df_nybb, con=engine, name=table, if_exists="append")
```

Comment:
Shall we read it back and validate to make sure that roundtrip works?

Reply:
Yeah we can do that. Not needed necessarily as roundtrips are tested in other tests but it does not harm.